### PR TITLE
Add filter for docs list items

### PR DESCRIPTION
**Pull request by laviro**
https://user-images.githubusercontent.com/26363699/125588323-f4d3a1c5-5063-4270-b9ed-2342ab392d71.mp4

### CODE REVIEW

**Review by melcorr — COMMENTED** [review]
Hey @laviro this looks great. However I think it would be a lot more powerful if we could use it for more than just the front matter and release notes. Do you think it would be possible that we could add it to the guides themselves?

**Comment by laviro** [comment]
hey @melcorr, sure
if you could point me to the HTML file that hosts the guides we could inject something like:
```js
addPageFilter('#content', ['.sect2 ul li'])
```

**Comment by laviro** [comment]
And forgot to mention.. i get 404 errors trying to reach the guides locally, maybe we should do it as a follow up?

**Comment by melcorr** [comment]
@lzap knows the magic of this hugo build best. He might know, but no rush with this!

File: web/static/index_assets/filter.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+function addPageFilter(refSelector, selectorToFilter) {
+    var ref = $(refSelector)[0];
+    if (!ref) return;
```
[the remark this answers is not in the document]
Comment (laviro) [reply web/static/index_assets/filter.js:3]: > `$(refSelector)` can return `null` and `null[0]` will fail.

magically, jQuery seem to always return something, just tested on my web console:
```js
var ref = $('#doesnt-exist')
// undefined
ref
// init {context: document, selector: "#doesnt-exist"}
ref[0]
// undefined
```

but I can add a few life-savers

[the remark this answers is not in the document]
Comment (ekohl) [reply web/static/index_assets/filter.js:3]: It appears you're right. It always returns some object. Then you can disregard my suggestion.

[the remark this answers is not in the document]
Comment (laviro) [reply web/static/index_assets/filter.js]: just pushed a small safe-check as we spoke

**Comment by laviro** [comment]
let's do the guides as a follow-up? @melcorr @lzap ?

**Comment by lzap** [comment]
I do not like this honestly, every browser has ctrl+f feature or similar. This is a code we need to maintain.

For the guides which are generated via asciidoc, this cannot be added easily. Only HTML header can be modified, but not anything in the body.

What is the hugo error you see? You need to start a hugo server to serve the content, I am not following.

**Comment by laviro** [comment]
> For the guides which are generated via asciidoc, this cannot be added easily. Only HTML header can be modified, but not anything in the body.

can we add a JS script there that amend the body?

> What is the hugo error you see? You need to start a hugo server to serve the content, I am not following.

I couldn't generate the guides that time

closing it for now as it seems not very useful here?
let's open another PR for the guides if we decide it's needed,
thanks all!

**Comment by lzap** [comment]
I am not blocking this, it is my personal preference, however, I can help you with building with hugo - ping me if you still want to continue.

**Pull request closed without merging**